packages/next/shared/lib/router/router.ts

```diff
@@ -437,7 +437,10 @@ function fetchRetry(
     // https://github.com/github/fetch#caveats
     credentials: 'same-origin',
     method: options.method || 'GET',
-    headers: options.headers ?? {},
+    headers: {
```

Comment:
Maybe `Object.assign` to have less polyfills?

Reply:
Updated in https://github.com/vercel/next.js/pull/37574/commits/c65c67f2628f52374163128f182e45222a42aaa8